packages/editor/src/components/document-bar/index.js

```diff
@@ -48,51 +48,32 @@ const icons = {
 };
 
 export default function DocumentBar() {
-	const {
-		isEditingTemplate,
-		templateId,
-		postType,
-		postId,
-		goBack,
-		getEditorSettings,
-	} = useSelect( ( select ) => {
+	const { postType, postId, goBack } = useSelect( ( select ) => {
 		const {
-			getRenderingMode,
-			getCurrentTemplateId,
 			getCurrentPostId,
 			getCurrentPostType,
 			getEditorSettings: getSettings,
 		} = select( editorStore );
-		const _templateId = getCurrentTemplateId();
 		const back = getSettings().goBack;
 		return {
-			isEditingTemplate:
-				!! _templateId && getRenderingMode() === 'template-only',
-			templateId: _templateId,
 			postType: getCurrentPostType(),
 			postId: getCurrentPostId(),
 			goBack: typeof back === 'function' ? back : undefined,
 			getEditorSettings: getSettings,
 		};
 	}, [] );
 
-	const { setRenderingMode } = useDispatch( editorStore );
-
 	const handleOnBack = () => {
-		if ( isEditingTemplate ) {
-			setRenderingMode( getEditorSettings().defaultRenderingMode );
-			return;
-		}
```

Comment:
This seems like the one place where we need a `goBack` method in the site editor's `usePostLinkProps`, and only when you have navigated from a page to edit its template. Still trying to figure out how best to handle this 🤔